### PR TITLE
Add sign-up flow to App studio

### PR DESCRIPTION
**Pull request by divyanshiGupta**
## Fixes 

Fixes: https://issues.redhat.com/browse/HAC-1021

## Description

This PR adds the signup flow to App Studio.

## Type of change

- [x] Feature

## Screen shots / Gifs for design review 

Signup page:
Screenshot 2022-04-25 at 4 27 19 PM

Screenshot 2022-04-25 at 4 27 27 PM

Waiting for approval page
Screenshot 2022-04-25 at 4 28 32 PM
Screenshot 2022-04-25 at 4 28 40 PM

## Browser conformance: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

 ## ToDo:
- [ ] Add tests

### CODE REVIEW

File: src/components/Signup/SignupBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import { CheckIcon } from '@patternfly/react-icons/dist/js/icons';
+
+import './SignupBanner.scss';
+
+const SignupBanner: React.FC = () => {
+  return (
+    <Card className="signup-banner" isLarge>
+      <CardTitle>Get Started with Red Hat App Studio</CardTitle>
+      <CardBody>
+        Red Hat App Studio is a fully managed experience that makes easier and faster for developers
```
Comment (rottencandy) [inline src/components/Signup/SignupBanner.tsx:12]:
```suggestion
        Red Hat App Studio is a fully managed experience that makes it easier and faster for developers
```

File: src/components/Signup/SignupView.tsx [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+                        App Studio provides the ability to run multiple instances of distributed
+                        microservices across geos and cloud platforms. Each application component
+                        can be set up for auto-scaling and fault tolerance.
+                      </CardBody>
+                    </Card>
+                  </SplitItem>
+                </Split>
+              </CardBody>
+              <CardFooter>
+                <ExternalLink href={'https://developers.redhat.com/'}>
```
Comment (rottencandy) [inline src/components/Signup/SignupView.tsx:173]: Nit:
```suggestion
                <ExternalLink href="https://developers.redhat.com/">
```

File: src/components/Signup/SignupView.tsx [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+      .catch();
+  };
+
+  if (!loaded) {
+    return (
+      <Bullseye>
+        <Spinner />
+      </Bullseye>
+    );
+  }
```
Comment (rottencandy) [inline src/components/Signup/SignupView.tsx:56]: Would it be a good idea to handle this in NamespacedPage itself? The check won't have to be repeated again that way. And `loaded` prop will also not be needed.

File: src/hooks/useSignupStatus.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
+import { Dispatch, SetStateAction, useEffect, useState } from 'react';
+import { commonFetch } from '@openshift/dynamic-plugin-sdk-utils';
+
+export enum UserSignupStatus {
+  NOT_SIGNEDUP = 'NotSignedup',
+  PENDING_APPROVAL = 'PendingApproval',
+  PROVISIONED = 'Provisioned',
+}
+
+export const useSignupStatus = (): [string, Dispatch<SetStateAction<string>>, boolean] => {
```
Comment (rottencandy) [inline src/hooks/useSignupStatus.ts:10]: Tiny nit: just for stronger typing
```suggestion
export const useSignupStatus = (): [UserSignupStatus, Dispatch<SetStateAction<string>>, boolean] => {
```

**Comment by rohitkrai03** [comment]
cc: @beaumorley @serenamarie125

**Review by rohitkrai03 — APPROVED** [review]
Tried it locally with new user. Works as expected.

https://user-images.githubusercontent.com/6041994/165297544-48975b95-fd05-41f5-9b1c-b2d0f06863f9.mov

/approve
/lgtm

**Pull request merged**